projects/packages/woocommerce-analytics/src/class-wc-analytics-tracking.php

```diff
@@ -152,7 +166,131 @@ private static function record_ch_event( $properties ) {
 		if ( is_wp_error( $event_obj->error ) ) {
 			return $event_obj->error;
 		}
-		return WC_Tracks_Client::record_event( $event_obj ); // phpcs:ignore WordPress.NamingConventions.ValidFunctionName.MethodNameInvalid @phan-suppress-current-line PhanTypeMismatchArgumentProbablyReal
+
+		return self::record_event_pixel( $event_obj );
+	}
+
+	/**
+	 * Record an event pixel using batching.
+	 *
+	 * @param WC_Tracks_Event|WC_Analytics_Ch_Event $event The event object.
+	 * @return bool|WP_Error True for success or WP_Error if the event pixel could not be fired.
+	 */
+	private static function record_event_pixel( $event ) {
+		$pixel = $event->build_pixel_url();
+
+		if ( ! $pixel ) {
+			return new WP_Error( 'invalid_pixel', 'cannot generate tracks pixel for given input', 400 );
+		}
+
+		// Queue the pixel and send on shutdown.
+		self::queue_pixel_for_batch( $pixel );
+		return true;
+	}
+
+	/**
+	 * Queue a pixel URL for batch sending.
+	 *
+	 * @param string $pixel The pixel URL to queue.
+	 */
+	private static function queue_pixel_for_batch( $pixel ) {
+		self::$pixel_batch_queue[] = $pixel;
+
+		// Register shutdown hook once.
+		if ( ! self::$shutdown_hook_registered ) {
+			add_action( 'shutdown', array( __CLASS__, 'send_batched_pixels' ), 1 );
+			self::$shutdown_hook_registered = true;
+		}
+	}
+
+	/**
+	 * Send all queued pixels using batched non-blocking requests.
+	 * This runs on the shutdown hook to batch all requests together.
+	 *
+	 * Uses Requests library's request_multiple() when available for true parallel
+	 * batching via curl_multi, otherwise falls back to sequential non-blocking requests.
+	 */
+	public static function send_batched_pixels() {
+		if ( empty( self::$pixel_batch_queue ) ) {
+			return;
+		}
+
+		// Add request timestamp and nocache to all pixels.
+		$pixels_to_send = array();
+		foreach ( self::$pixel_batch_queue as $pixel ) {
+			$pixels_to_send[] = WC_Tracks_Client::add_request_timestamp_and_nocache( $pixel );
+		}
+
+		// Try to use Requests library for true parallel batching.
+		$can_batch = ( class_exists( 'WpOrg\Requests\Requests' ) && method_exists( 'WpOrg\Requests\Requests', 'request_multiple' ) )
+			|| ( class_exists( 'Requests' ) && method_exists( 'Requests', 'request_multiple' ) );
+
+		if ( $can_batch ) {
```

Comment:
Would it make sense to check `can_batch` before adding to the pixel queue? For example, if `can_batch` is true, we add to the pixel queue; otherwise, we record the event as before.

Reply:
Addressed